qemu/target-i386/cpu.h

```diff
@@ -699,7 +699,7 @@ typedef enum {
 
 typedef struct SegmentCache {
     uint32_t selector;
-    target_ulong base;
+    uint64_t base;   /* handle 32 or 64 bit CPUs */
```

Comment:
on (3), i mean our struct in `x86.h` can just set "base" to *uint64_t*, but we leave the internal the struct "SegmentCache" of Qemu *unchanged*. then we can still copy the fields regardless how *target_ulong* is interpreted